Assets/5cript5/ContractScripts/ContractController.cs

```diff
@@ -4,27 +4,55 @@
 //Contract controller
 public class ContractController
 {
-    List<Contract> contracts;
+    Dictionary<string, Contract> contracts;
 
     public ContractController()
     {
-        this.contracts = new List<Contract>();
+        contracts = new Dictionary<string, Contract>();
     }
 
-    // Creates a contract and returns a pointer to it
-    public Contract CreateNewContract(string contractName, float timeToComplete, int assignedWorkers, int amountAwarded, int contractType)
+    //Creates a contract and return string id
+    public string CreateNewContract(string contractName, float timeToComplete, int assignedWorkers, int amountAwarded, int contractType)
     {
         Contract c = new Contract(contractName, timeToComplete, assignedWorkers, amountAwarded, contractType);
-        contracts.Add(c); 
-        return c;
+        contracts.Add(c.GetGuid(), c);
+        return c.GetGuid();
     }
 
+    //Removes a contract
+    public void RemoveContract(string guid)
+    {
+        contracts.Remove(guid);
+    }
+
+    //Return a contract
+    public Contract ReturnContract(string guid)
```

Comment:
Nitpick: maybe consider renaming ReturnContract() to GetContract to match the other function naming conventions :)